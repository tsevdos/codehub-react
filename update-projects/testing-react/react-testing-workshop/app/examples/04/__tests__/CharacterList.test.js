import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import CharacterList from "../CharacterList";

const data = [
  {
    name: "John Tsevdos",
    gender: "male"
  },
  {
    name: "Patricia Lebsack",
    gender: "female"
  },
  {
    name: "Dennis Schulist",
    gender: "male"
  },
];

describe("<CharacterList />", () => {

  it("renders correctly with snapshot (shallow)", () => {
    const wrapper = shallow(<CharacterList data={data} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders correctly with snapshot (mount)", () => {
    const wrapper = mount(<CharacterList data={data} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
