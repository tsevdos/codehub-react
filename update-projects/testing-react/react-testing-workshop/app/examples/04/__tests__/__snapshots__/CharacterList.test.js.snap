// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterList /> renders correctly with snapshot (mount) 1`] = `
<CharacterList
  data={
    Array [
      Object {
        "gender": "male",
        "name": "John Tsevdos",
      },
      Object {
        "gender": "female",
        "name": "Patricia Lebsack",
      },
      Object {
        "gender": "male",
        "name": "Dennis Schulist",
      },
    ]
  }
  onSelectCharacter={[Function]}
>
  <div>
    <h2
      className="is-size-4"
    >
      Unit Testing
    </h2>
    <h3
      className="is-size-5"
    >
      04: Snapshots testing
    </h3>
    <br />
    <ul
      id="characters"
    >
      <li
        key="John Tsevdos"
        onClick={[Function]}
      >
        John Tsevdos
         is 
        male
      </li>
      <li
        key="Patricia Lebsack"
        onClick={[Function]}
      >
        Patricia Lebsack
         is 
        female
      </li>
      <li
        key="Dennis Schulist"
        onClick={[Function]}
      >
        Dennis Schulist
         is 
        male
      </li>
    </ul>
  </div>
</CharacterList>
`;

exports[`<CharacterList /> renders correctly with snapshot (shallow) 1`] = `
<div>
  <h2
    className="is-size-4"
  >
    Unit Testing
  </h2>
  <h3
    className="is-size-5"
  >
    04: Snapshots testing
  </h3>
  <br />
  <ul
    id="characters"
  >
    <li
      key="John Tsevdos"
      onClick={[Function]}
    >
      John Tsevdos
       is 
      male
    </li>
    <li
      key="Patricia Lebsack"
      onClick={[Function]}
    >
      Patricia Lebsack
       is 
      female
    </li>
    <li
      key="Dennis Schulist"
      onClick={[Function]}
    >
      Dennis Schulist
       is 
      male
    </li>
  </ul>
</div>
`;
